pycsw/core/util.py

```diff
@@ -220,10 +220,39 @@ def bbox2wktpolygon(bbox):
 
     if bbox.startswith('ENVELOPE'):
         bbox = wktenvelope2bbox(bbox)
-    minx, miny, maxx, maxy = [float(coord) for coord in bbox.split(",")]
-    return 'POLYGON((%.2f %.2f, %.2f %.2f, %.2f %.2f, %.2f %.2f, %.2f %.2f))' \
+    minx, miny, maxx, maxy = [format_coord(float(coord)) for coord in bbox.split(",")]
+    wktGeometry = 'POLYGON((%s %s, %s %s, %s %s, %s %s, %s %s))' \
         % (minx, miny, minx, maxy, maxx, maxy, maxx, miny, minx, miny)
+    return wktGeometry
 
+def format_coord(number):
```

Comment:
I think this function can be replaced by `f"{number:.{precision}f}""`. This supports negative values too.